pkg/controller/bundlec/resource_sync_task.go

```diff
@@ -151,6 +151,15 @@ func (st *resourceSyncTask) processResource(res *smith_v1.Resource) resourceInfo
 	// Create or update resource
 	resUpdated, retriable, err := st.createOrUpdate(spec, actual)
 	if err != nil {
+		if api_errors.IsInvalid(err) {
```

Comment:
We may wrap the error later and that will break this logic. Would be good to add a test for this case and change the code to extract the cause like this: `if api_errors.IsInvalid(errors.Cause(err)) {`.